vcpkg.json

```diff
@@ -1,16 +1,23 @@
 {
-    "name": "azure-sdk-for-cpp",
-    "version-string": "1.0.0",
-    "dependencies": [
-        {
-            "name": "curl"
-        },
-        {
-            "name": "libxml2",
-            "platform": "!windows & !uwp"
-        },
-        {
-            "name": "openssl"
-        }
-    ]
+  "name": "azure-sdk-for-cpp",
+  "version-string": "1.0.0",
+  "builtin-baseline": "f0aa678b7471497f1adedcc99f40e1599ad22f69",
```

Comment:
I wish there were better docs for this.
I see this: https://github.com/microsoft/vcpkg/blob/master/docs/users/versioning.md#builtin-baseline
But I don't fully understand what's happening.
Is this a VCPKG repo's commit Id?  or an azure-sdk-cpp's repo commit id?  Or is this a openssl repo's commit id??

If this is a VCPKG repo's commit id, is my local vcpkg folder going to checkout that commit id before building the dependencies?

What would it happen if we have an Azure package (imagine azure-attestation-cpp) which requires azure-core-cpp with a version that is in a more recent VCPKG commit id than the one we set here for OpenSSL??   Isn't that a conflict?

Reply:
It took me some time to figure it out. It's a vcpkg repo commit ID, and it takes the place of the vcpkg-commit.txt. Basically it establishes a baseline in vcpkg against which all packages will be pulled.

With this added, I believe we can remove support for vcpkg-commit.txt from our CI pipeline, but that's out of scope for this pull request.

Reply:
Oh, and as best as I can figure, the way this works is that when vcpkg is executing the manifest base vcpkg install, it syncs a separate directory to that commit hash, so it doesn't affect your local vcpkg repo. However it does mean that your local vcpkg repo needs to have the commit hash in its history.